pkg/master/reconcilers/mastercount.go

```diff
@@ -151,13 +151,15 @@ func (r *masterCountEndpointReconciler) RemoveEndpoints(serviceName string, ip n
 
 	// Remove our IP from the list of addresses
 	new := []corev1.EndpointAddress{}
-	for _, addr := range e.Subsets[0].Addresses {
-		if addr.IP != ip.String() {
-			new = append(new, addr)
+	if len(e.Subsets) > 0 {
```

Comment:
if length is 0, can simply return, no action is required to remove the endpoints

Reply:
are there other places in this reconciler or in others that assume non-zero length subsets?

Reply:
please add a test case to cover the bug being fixed

Reply:
I searched reconciler related code and found one other place where check should be added.

Reply:
@liggitt 
I have added a test case.

Reply:
```
func checkEndpointSubsetFormatWithLease(e *corev1.Endpoints, expectedIPs []string, ports []corev1.EndpointPort, reconcilePorts bool) (formatCorrect bool, ipsCorrect bool, portsCorrect bool) {
	if len(e.Subsets) != 1 {
```
If  checkEndpointSubsetFormatWithLease returns false, e.Subsets would be filled starting line 200:
```
	if !formatCorrect {
		// Something is egregiously wrong, just re-make the endpoints record.
		e.Subsets = []corev1.EndpointSubset{{
```